test/unit/esys-policy-ac-sendselect.c

```diff
@@ -0,0 +1,156 @@
+                           TSS2_TCTI_POLL_HANDLE * handles,
+                           size_t * num_handles);
+    TSS2_RC(*setLocality) (TSS2_TCTI_CONTEXT * tctiContext, uint8_t locality);
+} TSS2_TCTI_CONTEXT_FAKE;
+
+static TSS2_RC
+tcti_fake_policy_ac_sendselect_transmit (
+    TSS2_TCTI_CONTEXT *tcti_ctx,
+    size_t size,
+    const uint8_t *cmd_buf)
+{
+    UNUSED(tcti_ctx);
+
+    TSS2_RC rc;
+    tpm_header_t header;
+    rc = header_unmarshal (cmd_buf, &header);
+    if (rc != TSS2_RC_SUCCESS) {
+        return rc;
+    }
+
+    assert_int_equal(header.code, TPM2_CC_Policy_AC_SendSelect);
+
+    return TSS2_RC_SUCCESS;
+}
+
+static TSS2_RC
+tcti_fake_policy_ac_sendselect_receive (
+    TSS2_TCTI_CONTEXT * tctiContext,
+    size_t * response_size,
+    uint8_t * response_buffer, int32_t timeout)
+{
+    UNUSED(tctiContext);
+    UNUSED(response_size);
+    UNUSED(response_buffer);
+    UNUSED(timeout);
+
+    return TSS2_RC_SUCCESS;
+}
+
+static void
+tcti_fake_policy_ac_sendselect_finalize (TSS2_TCTI_CONTEXT * tctiContext)
+{
+    UNUSED(tctiContext);
+}
+
+static TSS2_RC
+tcti_fake_initialize(TSS2_TCTI_CONTEXT * tctiContext, size_t * contextSize)
+{
+    TSS2_TCTI_CONTEXT_FAKE *tcti_fake =
+        (TSS2_TCTI_CONTEXT_FAKE *) tctiContext;
+
+    if (tctiContext == NULL && contextSize == NULL) {
+        return TSS2_TCTI_RC_BAD_VALUE;
+    } else if (tctiContext == NULL) {
+        *contextSize = sizeof(*tcti_fake);
+        return TSS2_RC_SUCCESS;
+    }
+
+    /* Init TCTI context */
+    memset(tcti_fake, 0, sizeof(*tcti_fake));
+    TSS2_TCTI_MAGIC(tctiContext) = TCTI_FAKE_MAGIC;
+    TSS2_TCTI_VERSION(tctiContext) = TCTI_FAKE_VERSION;
+    TSS2_TCTI_TRANSMIT(tctiContext) = tcti_fake_policy_ac_sendselect_transmit;
+    TSS2_TCTI_RECEIVE(tctiContext) = tcti_fake_policy_ac_sendselect_receive;
+    TSS2_TCTI_FINALIZE(tctiContext) = tcti_fake_policy_ac_sendselect_finalize;
+    TSS2_TCTI_CANCEL(tctiContext) = NULL;
+    TSS2_TCTI_GET_POLL_HANDLES(tctiContext) = NULL;
+    TSS2_TCTI_SET_LOCALITY(tctiContext) = NULL;
+
+    return TSS2_RC_SUCCESS;
+}
+
+static int
+setup(void **state)
+{
+    TSS2_RC r;
+    ESYS_CONTEXT *ectx;
+    size_t size = sizeof(TSS2_TCTI_CONTEXT_FAKE);
+    TSS2_TCTI_CONTEXT *tcti = malloc(size);
+
+    r = tcti_fake_initialize(tcti, &size);
+    if (r)
+        return (int)r;
+    r = Esys_Initialize(&ectx, tcti, NULL);
+    *state = (void *)ectx;
+    return (int)r;
+}
+
+static int
+teardown(void **state)
+{
+    TSS2_TCTI_CONTEXT *tcti;
+    ESYS_CONTEXT *ectx = (ESYS_CONTEXT *) * state;
+    Esys_GetTcti(ectx, &tcti);
+    Esys_Finalize(&ectx);
+    free(tcti);
+    return 0;
+}
+
+static void
+test_policy_ac_sendselect(void **state)
+{
+    //TSS2_RC r;
+    ESYS_CONTEXT *ectx = (ESYS_CONTEXT *) * state;
+
+    Esys_Policy_AC_SendSelect(ectx, ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
+                              NULL, NULL, NULL, 0);
+
+    //assert_int_equal(r, TSS2_RC_SUCCESS);
```

Comment:
add the return code buffer here as described above and check RC. This ensures the plumbing for responses is somewhat sane.